src/validation.cpp

```diff
@@ -2811,6 +2828,88 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
             cache.EraseStoredVariables(static_cast<uint32_t>(pindex->nHeight));
         }
 
+        if (pindex->nHeight == chainparams.GetConsensus().GreatWorldHeight) {
+            auto balance = cache.GetBalance(chainparams.GetConsensus().foundationShareScript, DCT_ID{0});
+            if (balance.nValue > 0) {
+                cache.SubBalance(chainparams.GetConsensus().foundationShareScript, balance);
+                cache.AddCommunityBalance(CommunityAccountType::CommunityDevFunds, balance.nValue);
+            }
+        }
+
+        if (pindex->nHeight >= chainparams.GetConsensus().GreatWorldHeight)
+        {
+            std::set<uint256> activeMasternodes;
+            cache.ForEachCycleProp([&](CPropId const & propId, CPropObject const & prop) {
+
+                if (activeMasternodes.empty()) {
+                    cache.ForEachMasternode([&](uint256 const & mnId, CMasternode node) {
+                        if (node.IsActive(pindex->nHeight) && node.mintedBlocks) {
+                            activeMasternodes.insert(mnId);
+                        }
+                        return true;
+                    });
+                    if (activeMasternodes.empty()) {
+                        return false;
+                    }
+                }
+
+                uint32_t voteYes = 0, voters = 0;
+                cache.ForEachPropVote([&](CPropId const & pId, uint8_t cycle, uint256 const & mnId, CPropVoteType vote) {
+                    if (pId != propId || cycle != prop.cycle) {
+                        return false;
+                    }
+                    if (activeMasternodes.count(mnId)) {
+                        ++voters;
+                        if (vote == CPropVoteType::VoteYes) {
+                            ++voteYes;
+                        }
+                    }
+                    return true;
+                }, CMnVotePerCycle{propId, prop.cycle});
+
+                auto valid = lround(voters * 10000.f / activeMasternodes.size()) >= chainparams.GetConsensus().props.minVoting;
+                if (valid) {
+                    uint32_t majorityThreshold;
+                    switch(prop.type) {
+                        case CPropType::CommunityFundRequest:
+                            majorityThreshold = chainparams.GetConsensus().props.cfp.majorityThreshold;
+                            break;
+                        case CPropType::BlockRewardRellocation:
+                            majorityThreshold = chainparams.GetConsensus().props.brp.majorityThreshold;
+                            break;
+                        case CPropType::VoteOfConfidence:
+                            majorityThreshold = chainparams.GetConsensus().props.voc.majorityThreshold;
+                            break;
+                        default:
+                            assert(false);
+                    }
+                    auto approved = lround(voteYes * 10000.f / voters) >= majorityThreshold;
```

Comment:
Pink paper says simple majority required for CFP, so yes vote should be more than 50%, not equal to it. 50% would be an equal split of yes and no and should not side with a yes vote under this circumstance. The pink paper says the same for VoC and block reallocation, that it should be more than 66.67% and not equal, though in this case more than or equal is less disagreeable.